mlir/lib/Target/IRDLToCpp/IRDLToCpp.cpp

```diff
@@ -472,31 +502,69 @@ void {0}::build(::mlir::OpBuilder &odsBuilder, ::mlir::OperationState &odsState,
-          return results.emitError(
-              "IRDL C++ translation only supports irdl.any "
-              "constraint for types");
-        }
-      }
-    }
-  }
+  if (failed(isSnakeCase(dialect.getSymName(), dialect)))
+    return failure();
 
-  return success();
+  LogicalResult res = success();
+  dialect.walk([&](mlir::Operation *op) {
+    res =
+        llvm::TypeSwitch<Operation *, LogicalResult>(op)
+            .Case<irdl::AttributeOp>([](irdl::AttributeOp op) -> LogicalResult {
+              return op.emitError(
+                  "IRDL C++ translation does not yet support attributes.");
+            })
+            .Case<irdl::AttributesOp>(
+                [](irdl::AttributesOp op) -> LogicalResult {
+                  return op.emitError(
+                      "IRDL C++ translation does not yet support attributes.");
+                })
+            .Case<irdl::AnyOfOp>([](irdl::AnyOfOp op) -> LogicalResult {
+              return op.emitError("IRDL C++ translation only supports irdl.any "
+                                  "constraint for types");
+            })
+            .Case<irdl::AllOfOp>([](irdl::AllOfOp op) -> LogicalResult {
+              return op.emitError("IRDL C++ translation only supports irdl.any "
+                                  "constraint for types");
+            })
+            .Case<irdl::BaseOp>([](irdl::BaseOp op) -> LogicalResult {
+              return op.emitError(
+                  "IRDL C++ translation does not yet support base types.");
+            })
+            .Case<irdl::ParametersOp>([](irdl::ParametersOp op)
+                                          -> LogicalResult {
+              return op.emitError(
+                  "IRDL C++ translation does not yet support type parameters.");
+            })
+            .Case<irdl::ParametricOp>(
+                [](irdl::ParametricOp op) -> LogicalResult {
+                  return op.emitError("IRDL C++ translation does not yet "
+                                      "support parametric operations.");
+                })
+            .Case<irdl::RegionOp>([](irdl::RegionOp op) -> LogicalResult {
+              return op.emitError(
+                  "IRDL C++ translation does not yet support regions.");
+            })
+            .Case<irdl::RegionsOp>([](irdl::RegionsOp op) -> LogicalResult {
+              return op.emitError(
+                  "IRDL C++ translation does not yet support regions.");
+            })
+            .Case<irdl::IsOp>([](irdl::IsOp op) -> LogicalResult {
+              return op.emitError("IRDL C++ translation only supports irdl.any "
+                                  "constraint for types");
+            })
+            .Case<irdl::CPredOp>([](irdl::CPredOp op) -> LogicalResult {
+              return op.emitError("IRDL C++ translation only supports irdl.any "
+                                  "constraint for types");
+            })
+            .Default(success());
```

Comment:
Do it the other way around, success if the operation is explicitly supported (and then check that the operands are constraints that you support, which you are not doing right now I believe).

Reply:
The reason I say this is because at some point I expect irdl.dialect will not be isolated from above anymore so SSA values could come from outside of it. This would hijack your checks.